packages/search/test/util/merge-sort/merge.test.ts

```diff
@@ -0,0 +1,946 @@
+    });
+
+    it("should be able to handle returning 0 when no data from a source is added", () => {
+      // Setup
+      const arrOne: number[] = Array.apply(null, Array(4)).map(() =>
+        random.number()
+      );
+      const arrTwo: number[] = Array.apply(null, Array(3)).map(() =>
+        random.number()
+      );
+      const arrThree: number[] = Array.apply(null, Array(2)).map(() =>
+        random.number()
+      );
+
+      const data: number[][] = [arrThree, arrOne, arrTwo];
+      const expected = [arrThree[0], arrOne[0]];
+
+      // Test
+      const result: { data: number[]; [key: string]: any } = kMerge(data, 2);
+
+      // Assertions
+      expect(result).toBeDefined();
+      expect(result).toEqual({ data: expected, "0": 1, "1": 1, "2": 0 });
+    });
+  });
+
+  describe("sorted k-way merges", () => {
+    it("should be able to handle 1 results set with data and 2 without", () => {
+      // Setup
+      const sortFunc = (a: number, b: number) => a - b;
+      const fakeArray: number[] = Array.apply(null, Array(15))
+        .map(() => random.number())
+        .sort(sortFunc);
+
+      const data: number[][] = [[], fakeArray, []];
+      const expected = [...fakeArray];
+
+      // Test
+      const cmptr = (one: number[], two: number[]) => {
+        if (one.length === 0) return two[0] ? 1 : 0;
+        if (two.length === 0) return one[0] ? -1 : 0;
+        return one[0] - two[0];
+      };
+      const result: { data: number[]; [key: string]: any } = kMerge(
+        data,
+        15,
+        cmptr
+      );
+
+      // Assertions
+      expect(result).toBeDefined();
+      expect(result).toEqual({ data: expected, "0": 0, "1": 15, "2": 0 });
+    });
+
+    it("should be able to handle 2 results set with data and 1 without", () => {
+      // Setup
+      const sortFunc = (a: number, b: number) => a - b;
+      const arrOne: number[] = Array.apply(null, Array(4))
+        .map(() => random.number())
+        .sort(sortFunc);
+      const arrTwo: number[] = Array.apply(null, Array(3))
+        .map(() => random.number())
+        .sort(sortFunc);
+
+      const data: number[][] = [[], arrOne, arrTwo];
+      const expected = [...arrOne].concat([...arrTwo]).sort(sortFunc);
+
+      // Test
+      const cmptr = (one: number[], two: number[]) => {
+        if (one.length === 0) return two[0] ? 1 : 0;
+        if (two.length === 0) return one[0] ? -1 : 0;
+        return one[0] - two[0];
+      };
+      const result: { data: number[]; [key: string]: any } = kMerge(
+        data,
+        10,
+        cmptr
+      );
+
+      // Assertions
+      expect(result).toBeDefined();
+      expect(result).toEqual({ data: expected, "0": 0, "1": 4, "2": 3 });
+    });
+
+    it("should be able to handle 3 results set with data", () => {
+      // Setup
+      const sortFunc = (a: number, b: number) => a - b;
+      const arrOne: number[] = Array.apply(null, Array(4))
```

Comment:
A more clear way to write this is with lodash, a la: `_.range(0, 4).map(...).sort(...)`

Reply:
Lodash currently isn't a dependency. If we decide to use it, I can certainly change this, as well as use other lodash functions in the actual implementation

Reply:
Oh right, I forgot that this is hub.js and lodash isn't included